### ISSUE TITLE
iCloud Sign-in not sticking

### ISSUE THREAD

**Issue by twinklebob** [saenzramiro/rambox#405]
I added iCloud mail a while back, with whichever version it first appeared, and it used to work.

But now whenever I sign in I see the icons of all the iCloud stuff appear, Mail gets selected and my inbox appears and then just as it looks about loaded it all disappears and I'm returned to the login screen. Does anyone else experience this?

Starting from the same address and using the same login details I can login fine if I use Chrome.

Here's what the console logs:

```
CloudOS:  No accountType value received from the widget
GET https://p09-ckdatabasews.icloud.com/database/1/com.apple.notes/production/p…3F-D0E9-42FB-8220-A35C148A65A4&clientMasteringNumber=16F86&dsid=1946572256 421 (Misdirected Request)
GET https://p09-contactsws.icloud.com/co/mecard/?clientBuildNumber=16FProject13…3F-D0E9-42FB-8220-A35C148A65A4&clientMasteringNumber=16F86&dsid=1946572256 421 (Client Error (421))
CloudOS:  The me-card was not retrieved. (The network request may have failed or a necessary service may be unavailable.)
Message Stat reporting throttled, not reporting.
GET https://p09-remindersws.icloud.com/rd/alarmtriggers/?clientBuildNumber=16FP…256&endDate=2016-10-11&lang=en&startDate=2016-10-10&usertz=Europe%2FLondon 421 (Unauthorized)
GET https://p09-contactsws.icloud.com/co/addressbook/?clientBuildNumber=16FHotf…MasteringNumber=16FHotfix3&dsid=1946572256&locale=en_US&order=last%2Cfirst 421 (Client Error (421))
GET https://p09-calendarws.icloud.com/ca/alarmtriggers/?clientBuildNumber=16FPr…256&endDate=2016-10-11&lang=en&startDate=2016-10-10&usertz=Europe%2FLondon 421 (Unauthorized)
```

Happy to provide more detail from the console logs if any of that looks useful.

## 

> Rambox 0.4.4
> Electron 1.4.1
> linux x64 3.19.0-32-generic

**Comment by jessicaxr1** [comment]
Having the same issue as you, on Windows 10/Chrome.

**Comment by LA5ERman** [comment]
Exactly same problem. Tried to switch off 2FA - same. It seems like Rambox is creating new session when loading Mail (only my guess). Rambox:
Version: 0.4.4
Platform: linux (x64)
Electron: 1.4.1
Chromium: 53.0.2785.113
Node: 6.5.0
OS: Ubuntu 16.04 LTS 64bit

**Comment by saenzramiro** [comment]
Seems now to be working OK here (Win10 x64).

Someone still have this issue?

**Comment by twinklebob** [comment]
I've still got it on 0.4.5. Is 0.5 going to be released soon with the certificate patch #460? Perhaps something else in that release fixes the problem with iCloud?

**Comment by saenzramiro** [comment]
I could reproduce this in 0.4.5, so seems to be a certificate error or a bug for Electron.

Tested in future 0.5.0 and works ok.
